### PR TITLE
Add validation to check non array field type

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Suject

Fixes:

## Checklist
- [ ] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: compiler-plugin-test/src/test/java/io/ballerina/stdlib/persist/compiler/plugin/CompilerPluginTest.java [hunk 1]
```diff
@@ -257,6 +257,26 @@ public void testUnSupportedFeature2() {
                 DiagnosticsCodes.PERSIST_120.getCode(), 1);
     }
 
+    @Test
+    public void testUnSupportedFeature3() {
+        DiagnosticResult diagnosticResult = loadPackage("package_24").getCompilation().diagnosticResult();
+        List<Diagnostic> errorDiagnosticsList = diagnosticResult.diagnostics().stream().
```
Comment (MohamedSabthar) [inline compiler-plugin-test/src/test/java/io/ballerina/stdlib/persist/compiler/plugin/CompilerPluginTest.java:263]: Seems this logic is repeated, we could abstract this into a function

File: compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/DiagnosticsCodes.java [hunk 2]
```diff
@@ -66,6 +66,8 @@ public enum DiagnosticsCodes {
             "used in another entity", ERROR),
     PERSIST_120("PERSIST_120", "unsupported features: array type is not supported yet",
             ERROR),
+    PERSIST_121("PERSIST_121", "unsupported features: {0} is not supported yet",
+            ERROR),
```
Comment (kaneeldias) [inline compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/DiagnosticsCodes.java:70]:
```suggestion
    PERSIST_120("PERSIST_120", "unsupported features: array type is not supported",
            ERROR),
    PERSIST_121("PERSIST_121", "unsupported features: {0} type is not supported",
            ERROR),
```

File: compiler-plugin-test/src/test/resources/test-src/sql_script/package_19/modules/foo/entities.bal [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import ballerina/persist;
```
Comment (daneshk) [inline compiler-plugin-test/src/test/resources/test-src/sql_script/package_19/modules/foo/entities.bal:1]: License header?

Comment (kalaiyarasiganeshalingam) [reply compiler-plugin-test/src/test/resources/test-src/sql_script/package_19/modules/foo/entities.bal:17]: Added in [2359de4](https://github.com/ballerina-platform/module-ballerina-persist/pull/50/commits/2359de46f1ea8c84cc7c0c481c91b059b75cae8f)

**Review by daneshk — APPROVED** [review]

**Pull request merged**